R/add_prop_ci.R

```diff
@@ -0,0 +1,168 @@
+#'     tbl_summary(statistic = all_categorical() ~ "{p}%",
+#'                 missing = "no") %>%
+#'     add_prop_ci() %>%
+#'     modify_table_styling(
+#'       columns = stat_0,
+#'       rows = !is.na(prop_ci_stat_0),
+#'       cols_merge_pattern = "{stat_0} ({prop_ci_stat_0})"
+#'     ) %>%
+#'     modify_footnote(everything() ~ NA)
+#' @section Example Output:
+#' \if{html}{Example 1}
+#'
+#' \if{html}{\figure{add_prop_ci_ex1.png}{options: width=50\%}}
+#'
+#' \if{html}{Example 2}
+#'
+#' \if{html}{\figure{add_prop_ci_ex2.png}{options: width=45\%}}
+add_prop_ci <- function(x, ...) {
+  UseMethod("add_prop_ci")
+}
+
+#' @rdname add_prop_ci
+#' @export
+add_prop_ci.tbl_summary <- function(x,
+                                    pattern = "{conf.low}%, {conf.high}%",
+                                    method = c("wilson", "wilson.no.correct",
+                                               "exact", "asymptotic"),
+                                    conf.level = 0.95,
+                                    ci_fun = NULL, ...) {
+  # resolving arguments --------------------------------------------------------
+  method <- match.arg(method)
+  if(!is.null(ci_fun)) ci_fun <- gts_mapper(ci_fun, "add_prop_ci(ci_fun=)")
+
+  # checking inputs ------------------------------------------------------------
+  if (!rlang::is_string(pattern))
+    stop("`pattern=` must be a string.", call. = FALSE)
+  updated_call_list <- c(x$call_list, list(add_prop_ci = match.call()))
+
+  # adding new column with CI --------------------------------------------------
+  x <-
+    x %>%
+    add_stat(
+      fns = all_categorical() ~ purrr::partial(single_prop_ci,
+                                               method = method,
+                                               conf.level = conf.level,
+                                               pattern = pattern,
+                                               ci_fun = ci_fun),
+      location = list(all_dichotomous() ~ "label", all_categorical(FALSE) ~ "level")
+    ) %>%
+    # moving the CI cols to after the original stat cols (when `by=` variable present)
+    # also renaming CI columns
+    modify_table_body(
+      function(.x) {
+        cols_to_order <-
+          .x %>%
+          select(all_stat_cols(), matches("^stat_\\d+_prop_ci$")) %>%
+          names()
+
+        .x %>%
+          dplyr::relocate(all_of(sort(cols_to_order)),
+                          .before = all_of(sort(cols_to_order)[1])) %>%
+          dplyr::rename_with(
+            .fn = ~paste0(
+              "prop_ci_",
+              stringr::str_replace(., pattern = "_prop_ci$", replacement = "")),
+            .cols = matches("^stat_\\d+_prop_ci$")
+          )
+      }
+    ) %>%
+    # updating CI column headers and footnotes
+    modify_header(matches("^prop_ci_stat_\\d+$") ~ paste0("**", conf.level*100, "% CI**")) %>%
+    modify_footnote(
+      update = matches("^prop_ci_stat_\\d+$") ~ "CI = Confidence Interval",
```

Comment:
I do not remember where translation is provided. Is the list of terms to be translated be automatically updated or should it be declared?